modules/schema/src/main/java/org/apache/ignite/internal/schema/marshaller/reflection/KvMarshallerImpl.java

```diff
@@ -118,7 +129,7 @@ public K unmarshalKey(Row row) throws MarshallerException {
     @Nullable
     @Override
     public V unmarshalValue(Row row) throws MarshallerException {
-        Object o = valMarsh.readObject(new RowReader(row, schema.keyColumns().size()), null);
+        Object o = valMarsh.readObject(new RowReader(row, valPositions), null);
```

Comment:
i'm wondering why have not we done the same for `unmarshalKey`? I bet it lack of tests on putAll

Reply:
Fixed.